### ISSUE TITLE
poetry 1.1.0b2 tries to install nltk...win32.exe on Linux

### ISSUE THREAD

**Issue by PAStheLoD** [python-poetry/poetry#2733]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Linux, Ubuntu 20.04 (Focal)
- **Poetry version**:  1.1.0b2 - on py2.7
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/PAStheLoD/1b1322ef8c43ffbd3b224ecd2cb51a85

## Issue

```
  • Installing nltk (3.4.4): Downloading... 100%
  • Installing nltk (3.4.4): Installing...
  • Installing nltk (3.4.4): Failed

  EnvCommandError

  Command ['/root/.cache/pypoetry/virtualenvs/raiker-yaOPvKXZ-py2.7/bin/pip', 'install', '--no-deps', '/root/.cache/pypoetry/artifacts/7d/34/7f/4de2547f7c07bf03f799034b9faf90e7c25be52d3d9846b1bfef08724a/nltk-3.4.4.win32.exe'] errored with the following return code 1, and output:
  DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
  ERROR: Invalid requirement: '/root/.cache/pypoetry/artifacts/7d/34/7f/4de2547f7c07bf03f799034b9faf90e7c25be52d3d9846b1bfef08724a/nltk-3.4.4.win32.exe'
  Hint: It looks like a path. It does exist.


  at ~/.poetry/lib/poetry/utils/env.py:915 in _run
       911│                 output = subprocess.check_output(
       912│                     cmd, stderr=subprocess.STDOUT, **kwargs
       913│                 )
       914│         except CalledProcessError as e:
    →  915│             raise EnvCommandError(e, input=input_)
       916│
       917│         return decode(output)
       918│
       919│     def execute(self, bin, *args, **kwargs):

```

The lock file has these lines:

```
nltk = [
    {file = "nltk-3.4.4.win32.exe", hash = "sha256:9819cbcaf38289a7f167cb9ef2f487f0c38f92b6866a8b1e63345b8d211f7709"},
    {file = "nltk-3.4.4.zip", hash = "sha256:764c20a5f8532a681c261af3c7d1a54768a35df6f3603df75e615cbd34e47cb5"},
]
```

I tried to change the order of the lines, or delete the win32 line, but the error still persists. I tried with other versions of "nltk". This used to work. (Last time the corresponding poetry.lock changed was ~4 months ago.)

Downgrading to 1.0.9 helps, even 1.1.0a2 works, but 1.1.0a3 doesn't.

**Comment by abn** [comment]
Thanks for the report @PAStheLoD. Looks like `nltk` has `bdist_wininst` which poetry was not handling correctly when choosing the archive to use. While these files are supported by PyPI, seems newer versions of nltk `3.5.5` do not use them either. As a workaround, you can switch to `3.5` for now. It should be something we will fix in the next pre-release.

Also, you should consider switching away from python 2.x. :)

**Comment by PAStheLoD** [comment]
Thanks for the quick reply!

(This is a project in-transition away from py2. But it's a sloooow process, due to lack of funds and strange over-engineered homegrown modules that never should have been created. Oh, and NLTK 3.5+ dropped py2 support, so we can't upgrade just yet. 😭 )

**Comment by abn** [comment]
I feel your pain!

As for a workaround you can also add it as a `url` dependency direct linking to the zip file on pypi. But, this should be fixed prior to `1.1.0` if you prefer to wait.

**Comment by PAStheLoD** [comment]
Thanks again!

Now that I know what the bug is, it's no problem to use 1.0.10 (`python2 -m virtualenv poetry.venv && poetry.venv/bin/pip install poetry==1.0.10` ... just as easy as using the get-poetry.py.)
